### ISSUE TITLE
mnesia_eleveldb cannot read  a  record when the key contain a map

### ISSUE THREAD

**Issue by YYChildren** [klarna/mnesia_eleveldb#13]
mnesia_eleveldb cannot read  a  record when the key contain a map.

This is a example:
```
1> mnesia:create_schema([erlang:node()]).
ok
3> mnesia:start().
5> mnesia_eleveldb:register().
{ok,leveldb_copies}
6> mnesia:create_table(db_ha, [{leveldb_copies, [node()]}, {record_name, r_kv}, {type, set}, {attributes, [k,v]}]).
15> mnesia:dirty_write(db_ha, {r_kv, #{a=>b}, 1}).
ok
16> mnesia:dirty_read(db_ha, #{a=>b}).            
** exception exit: {aborted,{badarg,[db_ha,#{a => b}]}}
     in function  mnesia:abort/1 (mnesia.erl, line 318)
20> Pattern = mnesia:table_info(db_ha, wild_pattern).
22> mnesia:dirty_match_object(db_ha, Pattern).
[{r_kv,1,#{a => b}},{r_kv,{1,1},1}]
```

**Comment by mikpe** [comment]
Confirmed.  With our internal versions of OTP + mnesia_ext and mnesia_eleveldb I can't even get the dirty_write to work, it doesn't fail but also doesn't put anything into the table.  Replacing the key with something trivial like the atom 'foo' makes it work.  Switching to disc_only_copies also makes it work.

I'll take a look next week, I think I know what the problem is.

**Comment by mikpe** [comment]
The issue that our copy of Ulf Wiger's "sext" library (https://github.com/uwiger/sext) doesn't have support for maps, and you're trying to put maps in the keys.  Upstream still doesn't fully support maps (there are 2 or 3 open PRs related to maps, with signs of backwards-compatibility being problematic).  I wouldn't want to bring maps into production only to face an incompatible format change after a while.

Bottom line: please ask upstream about maps support if you _really_ need it, otherwise please convert your tables to not contain maps.
